authorizer_lambda/src/authorizer_lambda/main.py

```diff
@@ -25,5 +67,15 @@ def handler(event, context) -> dict[str, Any]:
     authorization_request: TokenAuthorizationRequest = (
         TokenAuthorizationRequest.parse_obj(event)
     )
+    parsed_token = jwt.decode(
+        authorization_request.authorizationToken, options={"verify_signature": False}
+    )
+
+    is_cognito_token = parsed_token.get("sub")
```

Comment:
Its better to use the issuer in the `iss` field of the jwt token to verify where the token came from. That is also generally done when verifying JWT tokens, to make sure the token originated from the correct source. The [example dev token](https://github.com/araujoyuri/aws_terraform_interview/blob/80635551173bae53362790023d7e0fefe1221502/authorizer_lambda/tests/data/authorization_request_dev.json#L3) contanis `iss: dev.myapp.ai` and as [documented](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-the-access-token.html#user-pool-access-token-payload) in the link that is also in the README.md, AWS sets the issuer to `https://cognito-idp.<Region>.amazonaws.com/<your user pool ID>`. The user-pool-id` [is available in the config](https://github.com/araujoyuri/aws_terraform_interview/blob/80635551173bae53362790023d7e0fefe1221502/authorizer_lambda/src/authorizer_lambda/config.py#L16).

Reply:
Yeah, the first time I implemented this I did with a regex to check if it came from cognito, but in my mind would be easier to check for the `sub` property. It makes a lot more sense to check for the issuer though.